libsolidity/AST.h

```diff
@@ -415,6 +416,7 @@ class FunctionDefinition: public Declaration, public VariableScope, public Docum
 			getVisibility() >= Visibility::Internal;
 	}
 	virtual TypePointer getType(ContractDefinition const*) const override;
+	virtual bool isPartOfExternalInterface() const override { return isPublic() && m_isConstructor && getName().empty(); }
```

Comment:
Should be `isPublic() && !m_isConstructor && !getName().empty();` (two negations)